### PR TITLE
#568 Allow nodes to be disabled when a new or existing link is being …

### PR DESCRIPTION
**Pull request by tomlyn**
…dragged

Fixes: #568 

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: canvas_modules/common-canvas/src/common-canvas/svg-canvas-d3.js [hunk 1]
```diff
@@ -90,6 +90,7 @@ export default class SVGCanvasD3 {
 				this.config.enableDragWithoutSelect !== config.enableDragWithoutSelect ||
 				this.config.enableParentClass !== config.enableParentClass ||
 				this.config.enableHightlightNodeOnNewLinkDrag !== config.enableHightlightNodeOnNewLinkDrag ||
```
Comment (caritaou) [inline canvas_modules/common-canvas/src/common-canvas/svg-canvas-d3.js:92]: Should this be renamed as well? I see https://github.com/elyra-ai/canvas/issues/503 has been fixed and you're updating the examples in this PR

File: canvas_modules/common-canvas/src/common-canvas/svg-canvas-renderer.js [hunk 2]
```diff
@@ -3330,48 +3376,21 @@ export default class SVGCanvasRenderer {
...
 	setImageContent(imageSel, image) {
-		const nodeImageType = this.getImageType(image);
 		if (image !== imageSel.attr("data-image")) {
+			const nodeImageType = this.getImageType(image);
 			// Save image field in DOM object to avoid unnecessary image refreshes.
 			imageSel.attr("data-image", image);
 			if (nodeImageType === "svg") {
-				this.addCopyOfImageToDefs(image);
-				this.updateUseObject(imageSel, image);
+				d3.xml(image, { cache: "force-cache" }).then((data) => {
```
Comment (caritaou) [inline canvas_modules/common-canvas/src/common-canvas/svg-canvas-renderer.js:3385]: Are the images still stored in `defs`?

Comment (tomlyn) [reply canvas_modules/common-canvas/src/common-canvas/svg-canvas-renderer.js:3385]: No unfortunately with that approach we can only style icons once no matter how many times they are referenced from nodes. This means if we had two nodes both referring to the same icon we couldn't style one with fill = "grey" and one that is not grey. So now I include the SVG in the DOM for each node so in the example above the SVG would be inserted twice; once for each node. This means the icons can be styles separately on  a node by node basis.

File: canvas_modules/common-canvas/src/common-canvas/svg-canvas-d3.js (hunk 1 above)
Comment (tomlyn) [reply canvas_modules/common-canvas/src/common-canvas/svg-canvas-d3.js:92]: Yes, good catch. I found some other places with that typo. None will cause any runtime error but I fixed them nonetheless.

**Review by caritaou — APPROVED** [review]

**Pull request merged**